containers/utils/openimages/download.py

```diff
@@ -0,0 +1,156 @@
+
+        try:
+            os.mkdir("tar")
+        except FileExistsError:
+            pass
+        try:
+            os.mkdir("tar/train")
+        except FileExistsError:
+            pass
+        try:
+            os.mkdir("tar/test")
+        except FileExistsError:
+            pass
+        try:
+            os.mkdir("out")
+        except FileExistsError:
+            pass
+        for image_path in self.images:
+            image = cv2.imread(image_path)
+            height, width, channels = image.shape
+            file_id = image_path.split("/")[-1].rstrip(".jpg")
+            copyfile(image_path, "tar/" + image_path.split("/")[-1])
+            self.image_data.update({file_id: {"height": height, "width": width}})
+
+        all_labels_df = pandas.read_csv(os.path.join(self.directory, "train-annotations-bbox.csv"))
+        all_labels_df.set_index("ImageID", inplace=True)
+
+        labels = [i.lower() for i in self.labels]
+        for key in self.image_data.keys():
+            entry = all_labels_df.loc[key]
+            height = self.image_data[key]["height"]
+            width = self.image_data[key]["width"]
+            if type(entry) == pandas.DataFrame:
+                for row in entry.iterrows():
+                    box = {i.lower(): row[1][i] for i in ["XMin", "XMax", "YMin", "YMax"]}
+                    label = self.label_map[row[1]["LabelName"]].lower()
+                    if label in labels:
+                        print(label)
+                        self.parse_line(key + '.jpg', label, height, width, box)
+            else:
+                box = {i.lower(): entry.get(i) for i in ["XMin", "XMax", "YMin", "YMax"]}
+                label = self.label_map[entry.get("LabelName")].lower()
+                if label in labels:
+                    print(label)
+                    self.parse_line(key + '.jpg', label, height, width, box)
+
+        print("creating subfolders")
+        self.create_subfolders("train")
+        self.create_subfolders("test")
+
+    def create_subfolders(self, name):
+
+        with open("tar/{}/_annotations.csv".format(name), 'w+') as csv:
+            csv.write("filename,width,height,class,xmin,ymin,xmax,ymax\n")
+            r = range(len(self.csv))[:int(len(self.csv) * .7)] if name == "train" else range(len(self.csv))[
+                                                                                       int(len(self.csv) * .7):]
+            for i in r:
+                row = self.csv[i]
+                csv.write("{},{},{},{},{},{},{},{}\n".format(row["filename"], row["width"], row["height"], row["class"],
+                                                             row["xmin"], row["ymin"], row["xmax"], row["ymax"]))
+                copyfile("tar/" + row["filename"], "tar/" + name + '/' + row["filename"])
+                # self.label_frame("tar/" + row["filename"], row["class"], row["xmin"], row["xmax"], row["ymin"], row["ymax"])
```

Comment:
Do we need this line? If it is for debug, can we stick it in an if statement?

Reply:
will remove.